apps/condo/domains/news/schema/ExportNewsRecipientsService.js

```diff
@@ -0,0 +1,242 @@
+                const { data: { newsItemScopes, organizationId } } = args
+                const locale = extractReqLocale(context.req) || conf.DEFAULT_LOCALE
+
+                const organization = await Organization.getOne(context, {
+                    id: organizationId,
+                    deletedAt: null,
+                })
+                if (!organization) throw new GQLError(errors.WRONG_ORGANIZATION_ID, context)
+
+                let properties
+                if (!isEmpty(newsItemScopes)) {
+                    properties = newsItemScopes.map(newsItemScope => {
+                        if (!isNil(newsItemScope.property)) return newsItemScope.property.id
+                    })
+                }
+
+
+                let residentsByProperties
+                if (!isEmpty(properties)) {
+                    residentsByProperties = await loadListByChunks({
+                        context: context,
+                        list: Resident,
+                        chunkSize: 50,
+                        limit: 100000,
+                        where: {
+                            property: {
+                                id_in: properties,
+                            },
+                            deletedAt: null,
+                        },
+                    })
+                }
+
+                const recipientsByOrganization = []
+                if (organizationId && isEmpty(newsItemScopes)) {
+                    const residentsByOrganization = await Resident.getAll(context, {
+                        organization: {
+                            id: organizationId,
+                        },
+                        deletedAt: null,
+                    })
+
+                    const propertiesByOrganization = await loadListByChunks({
+                        context: context,
+                        list: Property,
+                        chunkSize: 50,
+                        limit: 100000,
+                        where: {
+                            organization: {
+                                id: organizationId,
+                            },
+                            deletedAt: null,
+                        },
+                    })
+
+                    for (let property of propertiesByOrganization) {
+                        const units = getUnitsFromProperty({ map: property.map })
+
+                        const recipientsData = units.reduce((acc, unit) => {
+                            if (residentsByOrganization.find((resident) => unit.unitName === resident.unitName)) {
+                                acc.push({ address: property.address, unitName: unit.unitName, hasResident: true })
+                                return [...acc]
+                            }
+                            acc.push({ address: property.address, unitName: unit.unitName, hasResident: false })
+                            return [...acc]
+                        }, [])
```

Comment:
you have repeat code below, maybe move it to a function?

Reply:
I think that in this case the readability of the code will not improve.